### ISSUE TITLE
Only account holder should be able to update their name details

### ISSUE THREAD

**Issue by alainna** [BirkbeckCTP/janeway#2576]
**Describe the bug**

**Janeway version**
Janeway preprints (1.4.0)

**To Reproduce**
1. Log in on moderator/staff/admin account
2. Navigate to a submission e.g. https://dev.eartharxiv.org/repository/manager/1298/ and note the names / email addresses of the authors. Choose one to do a sample update.
![image](https://user-images.githubusercontent.com/15632391/140057958-5db782e4-400f-484b-9bc8-bec0dce47876.png)

3. In Django admin, search the accounts for that author and open their entry
![image](https://user-images.githubusercontent.com/15632391/140057981-4e707c94-8d8e-456e-ae68-9b8e0e39286f.png)

4. Return to the preprint, click "Edit" next to the author's entry, update their name, and click "Save"
![image](https://user-images.githubusercontent.com/15632391/140058180-bb8fa6d7-d157-43c3-8523-f8c49044060f.png)

5. Refresh their entry and see the new name now on their account: 
![image](https://user-images.githubusercontent.com/15632391/140058342-7ef61ad9-1aeb-4a8e-ad7b-06fce48e4a9c.png)

6. Return to the preprint, click "Edit" next to the author's entry, update their name and email address, and click "Save" 
![image](https://user-images.githubusercontent.com/15632391/140058659-ffd742b8-b711-424d-a54a-d13e8fe67510.png)

7. Refresh their entry and see new name and email on their account; note that password has been reset: 
![image](https://user-images.githubusercontent.com/15632391/140060689-442226ed-468b-4175-bac6-e1768e76233c.png)

**Expected behavior**
1. If moderator updates author name / email address, then that version of the name/email is associated *only* with the preprint (or becomes an "author" profile that is linked to the account but separate from the account) and no updates occur on that author's account data. OR: 
2. Moderator cannot update author name / email address of an existing account, similar to submitting author. 

**Screenshots**

**Additional context**
This may require some rethinking of https://github.com/BirkbeckCTP/janeway/issues/2523

**Comment by tingletech** [comment]
FWIW, the fake `@eartharxiv` users I  think don't have passwords set / can't log in.

**Comment by alainna** [comment]
Good point @tingletech. The account I used for this one did have something in that field when I first loaded the page, but I didn't think to take a screenshot. However I also just tried to replicate this by updating the email address of an account that I controlled -- I couldn't replicate. So I'll edit the above to remove the reference to passwords being reset. 

(And I can confirm that I could successfully log into the account whose email address had been updated with the old email address (the one listed as the "acacount") and see the new name and email address in my author profile.)

**Comment by alainna** [comment]
We noticed when @justingonder was testing an Journal Editor workflow that Editors may also update user's email addresses when eg editing enrolled users.

**Comment by justingonder** [comment]
It might be nice to reserve editing of email addresses for 'is staff' #3374  'journal manager' or #3498 'Repository manager' roles. It just terrifies me a bit to have all editors making such important user account changes. 
@alainna

**Comment by ajrbyers** [comment]
For us this is highly philosophical. Janeway is built to empower Editors/Repo Managers. It’s one of the things they like about it. I don’t think there is any instance where either of these users would edit user data inc email address without good reason.

**Comment by joemull** [comment]
We discussed this in a backlog refinement meeting today. We would like to move toward using frozen author records for this functionality, like in the journals system in Janeway. This would require some reworking of the current repository workflow.

Ideally we'd want to avoid duplicating the frozen author model for the repository system, so we'd like to make the model generic for both journals and repositories.

**Comment by S-Haime** [comment]
Close in favour of #2449
